### PR TITLE
[mieb] Any2TextMultipleChoice Abstask&Evaluator & four tasks in CV-bench

### PR DESCRIPTION
**Pull request by gowitheflow-1998**
Adding:
1. Any2Text Multiple Choice Abstask and Evaluator 
2. four tasks from CV-bench that uses above (count, relation, depth, distance).

the logic is first grabbing unique candidate choices from the task (e.g., the 788 questions in the count task only has 17 unique candidates), encode them, store in a {candidate: embedding} dict. In scoring time, we query the dict and do similarity matching. This gets rid of the need to repetitive encoding, and makes it easier for queries with different number of candidates that can't be supported by `AbsTaskImageTextPairClassification` atm.

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

**Comment by gowitheflow-1998** [comment]
also, this and `Any2AnyRetrieval` can be extended to audio and video for moeb at some point I think!

File: mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+                "test": {
+                    # to do
+                }
+            },
+        },
+    )
+
+    def load_data(self, **kwargs):
+        # if self.data_loaded:
+        #     return
```
Comment (isaac-chung) [inline mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py:57]: Can this be removed?

File: mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        dataset={
+            "path": "nyu-visionx/CV-Bench",
+            "revision": "22409a927ab5cf68e3655023d51694587455fc99",
+        },
+        type="Any2TextMutipleChoice",
+        category="it2t",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2024-01-01", "2024-12-31"),
```
Comment (isaac-chung) [inline mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py:31]: Date on the arxiv pdf:
```suggestion
        date=("2024-01-01", "2024-06-24"),
```

File: mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+  title={placeholder},
+  author={placeholder and others},
+  journal={arXiv preprint arXiv:2401.13478},
+  year={2024}
+}""",
+        descriptive_stats={
+            "n_samples": {"test": 788},
+            "avg_character_length": {
+                "test": {
+                    # to do
```
Comment (isaac-chung) [inline mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py:49]: Just wondering if this was a private note.

File: mteb/evaluation/evaluators/Image/Any2TextMultipleChoiceEvaluator.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        for q_embedding, choice in tqdm(zip(query_embeddings, choices)):
+            choice_embeddings = torch.vstack(
+                [label_embedding_dict[c] for c in choice]
+            )  # (choice_size, embedding_dim)
+            q_embedding = q_embedding[np.newaxis, :]
+            cos_sim = cosine_similarity(q_embedding, choice_embeddings)
+            predictions.append(np.argmax(cos_sim))
+
+        metrics = {}
+        metrics["accuracy"] = accuracy_score(predictions, answers)
```
Comment (isaac-chung) [inline mteb/evaluation/evaluators/Image/Any2TextMultipleChoiceEvaluator.py:98]: nit: Could we combine this into one line?

**Review by isaac-chung — APPROVED** [review]
Great work with the AbsTasks! Mostly small non-blocking comments.

File: mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py (hunk 2 above)
Comment (gowitheflow-1998) [reply mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py]: have updated. Thanks!

File: mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py (hunk 3 above)
Comment (gowitheflow-1998) [reply mteb/tasks/Image/Any2TextMultipleChoice/eng/CVBench.py:49]: yes. Still need to decide what are the useful metadata for this new AbsTask and image datasets in general. Think we can solve this as an issue at last!

File: mteb/evaluation/evaluators/Image/Any2TextMultipleChoiceEvaluator.py (hunk 4 above)
Comment (gowitheflow-1998) [reply mteb/evaluation/evaluators/Image/Any2TextMultipleChoiceEvaluator.py:98]: not too sure if there are other useful metrics to be added in which case leaving as is might be easier?

Comment (isaac-chung) [reply mteb/evaluation/evaluators/Image/Any2TextMultipleChoiceEvaluator.py:98]: Okay that's fine

**Pull request merged**